src/manifolds/Tucker.jl

```diff
@@ -0,0 +1,884 @@
+Any tensor $p$ on the Tucker manifold can be represented as a multilinear product in HOSVD
+[^DeLathauwer2000] form
+```math
+p = (U_1,\dots,U_D) \cdot \mathcal{C}
+```
+where $\mathcal C \in \mathbb{F}^{R_1 \times \dots \times R_D}$ and, for $d=1,\dots,D$,
+the matrix $U_d \in \mathbb{F}^{N_d \times R_d}$ contains the singular vectors of the
+$d$th unfolding of $\mathcal{A}$
+
+# Tangent space
+
+The tangent space to the Tucker manifold at
+$p = (U_1,\dots,U_D) \cdot \mathcal{C}$ is [^Koch2010]
+```math
+T_p \mathcal{M} =
+\bigl\{
+(U_1,\dots,U_D) \cdot \mathcal{C}^\prime
++ \sum_{d=1}^D \bigl(
+    (U_1, \dots, U_{d-1}, U_d^\prime, U_{d+1}, \dots, U_D)
+    \cdot \mathcal{C}
+\bigr)
+\bigr\}
+```
+where $\mathcal{C}^\prime$ is arbitrary, $U_d^{\mathrm{H}}$ is the Hermitian adjoint of
+$U_d$, and $U_d^{\mathrm{H}} U_d^\prime = 0$ for all $d$.
+
+# Constructor
+    Tucker(N::NTuple{D, Int}, R::NTuple{D, Int}[, field = ℝ])
+
+Generate the manifold of `field`-valued tensors of dimensions  `N[1] × … × N[D]` and
+multilinear rank `R = (R[1], …, R[D])`.
+
+[^DeLathauwer2000]:
+    > Lieven De Lathauwer, Bart De Moor, Joos Vandewalle: "A multilinear singular value decomposition"
+    > SIAM Journal on Matrix Analysis and Applications, 21(4), pp. 1253-1278, 2000
+    > doi: [10.1137/S0895479896305696](https://doi.org/10.1137/S0895479896305696)
+
+[^Koch2010]:
+    > Othmar Koch, Christian Lubic, "Dynamical Tensor approximation"
+    > SIAM Journal on Matrix Analysis and Applications, 31(5), pp. 2360-2375, 2010
+    > doi: [10.1137/09076578X](https://doi.org/10.1137/09076578X)
+"""
+struct Tucker{N,R,D,𝔽} <: AbstractManifold{𝔽} end
+function Tucker(n⃗::NTuple{D,Int}, r⃗::NTuple{D,Int}, field::AbstractNumbers=ℝ) where {D}
+    @assert is_valid_mlrank(n⃗, r⃗)
+    return Tucker{n⃗,r⃗,D,field}()
+end
+
+#=
+HOSVD{T, D}
+
+Higher-order singular value decomposition of an order D tensor with eltype T
+fields:
+* U: singular vectors of the unfoldings
+* core: core tensor
+* σ : singular values of the unfoldings
+=#
+struct HOSVD{T,D}
+    U::NTuple{D,Matrix{T}}
+    core::Array{T,D}
+    σ::NTuple{D,Vector{T}}
+end
+
+@doc raw"""
+    TuckerPoint{T, D}
+
+An order `D` tensor of fixed multilinear rank and entries of type `T`, which makes it a
+point on the [`Tucker`](@ref) manifold. The tensor is represented in HOSVD form.
+
+# Constructors:
+    TuckerPoint(core::AbstractArray{T, D}, factors::Vararg{<:AbstractMatrix{T}, D}) where {T, D}
+
+Construct an order `D` tensor of element type `T` that can be represented as the
+multilinear product `(factors[1], …, factors[D]) ⋅ core`.
+It is assumed that the dimensions of the core are the multilinear
+rank of the tensor and that the matrices `factors` each have full rank. No further
+assumptions are made.
+
+    TuckerPoint(p::AbstractArray{T, D}, mlrank::NTuple{D, Int}) where {T, D}
+
+The low-multilinear rank tensor arising from the sequentially truncated the higher-order
```

Comment:
I might have missed this one: Could you explain T and D here, too? Should be similar to the line above, but just for completeness.